doc/_gallery/plot_3_1_4_frequency_resolution.py

```diff
@@ -25,6 +25,17 @@
 
 x = np.real(amgauss(128) * fmlin(128)[0])
 window = np.ones((128,))
-stft = ShortTimeFourierTransform(x, n_fbins=128, fwindow=window)
-stft.run()
+nr_samples = 128
+n_fbins = nr_samples
+nperseg = 128
+noverlap = nperseg - 1
+nfft = 128
```

Comment:
Pretty much all of these variables should be picked up by default.

Reply:
I have given these values explicitly to ensure that I reproduce the scenario in the Time-Frequenc Toolbox document as closely as I can. We could arrange for some of these to be picked up by default but I think that  hiding the values could lead to confusion, especially to newcomers to tftb (such as me).

According to the description of this issue, the ShortTimeFourierTransform should be a thin wrapper around Scipy's stft. Scipy's stft does provide default values but they are somewhat different to the values you are proposing.

I think it would be rather confusing to override Scipy's default values with new default values. It would also be very error prone. My preference  is to explicitly provide the parameters as this makes it much clearer what will actually be calculated.

Reply:
This is a good point - let me figure out how to set good defaults, and still display our deliberate choices for different parameters in this script.